app/models/recipe.rb

```diff
@@ -3,8 +3,19 @@ class Recipe < ApplicationRecord
   has_many :recipe_foods, dependent: :destroy
 
   validates :name, presence: true
-  validates :quantity, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
-  validates :preparatrion_time, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :preparation_time, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
   validates :cooking_time, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
   validates :description, presence: true
+
+  def self.public_list
+    Recipe.all.where(public: true)
+  end
+
+  def total_price
+    sum = 0
+    recipe_foods.each do |recipe_food|
+      sum += recipe_food.total_price
+    end
+    sum
+  end
```

Comment:
[OPTIONAL] For best practice please consider using a single line block for this method.

`recipe_foods.each { |recipe_food| sum += recipe_food.total_price }`